src/ts/utils/test/match.spec.ts

```diff
@@ -0,0 +1,72 @@
+import { IMatch } from "../..";
+import { mapThroughSkippedRanges } from "../match";
+
+describe("Match helpers", () => {
+  const getRuleMatch = (from: number, to: number): IMatch => ({
+    ruleId: "rule-id",
+    category: { id: "category-id", name: "Category", colour: "puce" },
+    from,
+    to,
+    matchId: "match-id",
+    precedingText: "",
+    subsequentText: "",
+    matchedText: "placeholder text",
+    message: "placeholder message",
+    matchContext: "[placeholder text]",
+    matcherType: "regex"
+  });
+
+  describe("mapMatchThroughSkippedRanges", () => {
+    it("should account for a range skipped before the given range", () => {
+      const ruleMatch = getRuleMatch(10, 15);
+      const skippedRange = [{ from: 0, to: 5 }];
+      const mappedMatch = mapThroughSkippedRanges(ruleMatch, skippedRange);
+      expect(mappedMatch.from).toBe(16);
+      expect(mappedMatch.to).toBe(21);
+    });
+
+    it("should account for a range skipped within the given range", () => {
+      const ruleMatch = getRuleMatch(10, 15);
+      const skippedRange = [{ from: 10, to: 15 }];
+      const mappedMatch = mapThroughSkippedRanges(ruleMatch, skippedRange);
+      expect(mappedMatch.from).toBe(16);
+      expect(mappedMatch.to).toBe(21);
+    });
+
+    it("should account for a range skipped within the given range, and extending beyond it", () => {
+      const ruleMatch = getRuleMatch(8, 12);
+      const skippedRange = [{ from: 8, to: 15 }];
```

Comment:
Do we account for a case where a skipped range would put two characters next to each other which would previously have been separated? I.e. ABC**DEF**G -> ABCG -> matches on CG. How do those ranges get mapped?

Reply:
This is I think accounted for in `should account for a range skipped within the given range` – we stretch the match across that range.

Reply:
(What the client does this that, at the moment, is another story – we drop the match completely, as it touches a note. This is a naive approach that works on the assumption that any ranges touched by user-defined marks (notes, flags) have been reviewed, and we may well want to review it down the line.)